### PR TITLE
Add table oci_audit_configuration. Closes #93

### LINKED ISSUE #93: Add table oci_audit_configuration
**References**
Add any related links that will help us understand the resource, including vendor documentation, related GitHub issues, and Go SDK documentation.

### PR DESCRIPTION
**Pull request by bigdatasourav**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
TEST: tests/oci_audit_configuration
Running terraform
oci_audit_configuration.named_test_resource: Creating...
oci_audit_configuration.named_test_resource: Creation complete after 6s [id=ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

Outputs:

retention_period_days = 365
tenancy_ocid = "ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq"

Running SQL query: test-list-query.sql
	
[
  {
    "retention_period_days": 365,
    "tenant_id": "ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql

null
✔ PASSED

Running SQL query: test-turbot-query.sql

[
  {
    "tenant_id": "ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq"
  }
]
✔ PASSED

POSTTEST: tests/oci_audit_configuration

TEARDOWN: tests/oci_audit_configuration

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
 select
  retention_period_days
from
  oci_audit_configuration;
+-----------------------+
| retention_period_days |
+-----------------------+
| 365                   |
+-----------------------+
```
</details>

### CODE REVIEW

File: oci-test/tests/oci_audit_configuration/test-list-query.sql [hunk 1]
```diff
@@ -0,0 +1,3 @@
+select retention_period_days,tenant_id
```
Comment (khushboo9024) [inline oci-test/tests/oci_audit_configuration/test-list-query.sql:1]:
```suggestion
select retention_period_days, tenant_id
```

File: oci-test/tests/oci_audit_configuration/test-not-found-query.sql [hunk 2]
```diff
@@ -0,0 +1,3 @@
+select retention_period_days,tenant_id
```
Comment (khushboo9024) [inline oci-test/tests/oci_audit_configuration/test-not-found-query.sql:1]:
```suggestion
select retention_period_days, tenant_id
```

File: oci/plugin.go [hunk 3]
```diff
@@ -43,6 +43,7 @@ func Plugin(ctx context.Context) *plugin.Plugin {
 			"oci_identity_user":                  tableIdentityUser(ctx),
 			"oci_objectstorage_bucket":           tableObjectStorageBucket(ctx),
 			"oci_region":                         tableIdentityRegion(ctx),
+			"oci_audit_configuration":            tableAuditConfiguration(ctx),
```
Comment (khushboo9024) [inline oci/plugin.go:46]: Sort the list.

File: oci/service.go [hunk 4]
```diff
@@ -230,6 +232,44 @@ func coreVirtualNetworkService(ctx context.Context, d *plugin.QueryData, region
 	return sess, nil
 }
 
+// coreVirtualNetworkService returns the service client for OCI Audit Service
```
Comment (khushboo9024) [inline oci/service.go:235]:
```suggestion
// auditService returns the service client for OCI Audit Service
```

**Review by khushboo9024 — CHANGES_REQUESTED** [review]
Please make suggested changes and resolve merge conflict.

File: oci/service.go [hunk 5]
```diff
@@ -553,8 +593,8 @@ func cloudGuardService(ctx context.Context, d *plugin.QueryData) (*session, erro
 	}
 
 	sess := &session{
-		TenancyID:        tenantID,
-		CloudGuardClient: client,
```
Comment (khushboo9024) [inline oci/service.go:557]: why we are removing `CloudGuardClient` ?

**Review by khushboo9024 — APPROVED** [review]
LGTM

**Comment by bigdatasourav** [comment]
can fetch retention period from https://github.com/turbot/steampipe-plugin-oci/issues/105

**Pull request closed without merging**